### PR TITLE
Refocus Code Editor after pressing ReplaceAll button

### LINKED ISSUE #70049: ESC key does not work sometimes in the script editor.
### Godot version

v3.5.1.stable.official [6fed1ffa3]

### System information

w10 64

### Issue description

Watch the video:
I want to replace a word and open the replacer.
I immediately decide that I won't replace it in the end and close the replacer by pressing ESC.
Again I reopen the replacer determined to replace.
I write the new word and click on "Replace all".
Now I press ESC but ESC doesn't close the replacer, ESC doesn't work now and I have to close the replacer using the mouse.

Conclusion:
The ESC key will not close the replacer when you click the "Replace All" button.

https://user-images.githubusercontent.com/19652075/207557994-2055896a-ffa1-492b-b28f-39720fe1ac87.mp4

### Steps to reproduce

Watch the video and the description.

### Minimal reproduction project

...

### PR DESCRIPTION
**Pull request by stmSi**
Fix: #70049 

for 3.x: #70521

### CODE REVIEW

**Comment by DissonantVoid** [comment]
shouldn't you also change the replace function to do the same? pressing the replace button also takes focus

**Comment by stmSi** [comment]
Oh right.

**Comment by stmSi** [comment]
done

File: editor/code_editor.cpp [hunk 1]
```diff
@@ -222,6 +222,11 @@ void FindReplaceBar::_replace() {
 	} else {
 		text_editor->deselect(0);
 	}
+
+	// or we could also check replace button focus // if(replace->has_focus()){
```
Comment (akien-mga) [inline editor/code_editor.cpp:226]: This kind of comment should go in the PR discussion, not in the code :)

Comment (stmSi) [reply editor/code_editor.cpp]: i remove comment now.

**Review by MewPurPur — APPROVED** [review]

File: editor/code_editor.cpp [hunk 2]
```diff
@@ -222,6 +222,10 @@ void FindReplaceBar::_replace() {
 	} else {
 		text_editor->deselect(0);
 	}
+
+	if (!replace_text->has_focus()) {
+		text_editor->grab_focus();
+	}
```
Comment (MewPurPur) [inline editor/code_editor.cpp:228]: You should make sure that the "Replace" button isn't selected either, not just the LineEdit.

**Comment by stmSi** [comment]
Looking at the code again after months, I just found much better solution for #70049 .
Can u check again @MewPurPur ?

File: editor/code_editor.cpp [hunk 3]
```diff
@@ -126,22 +126,18 @@ void FindReplaceBar::unhandled_input(const Ref<InputEvent> &p_event) {
 		return;
 	}
 
-	Control *focus_owner = get_viewport()->gui_get_focus_owner();
-	if (text_editor->has_focus() || (focus_owner && vbc_lineedit->is_ancestor_of(focus_owner))) {
```
Comment (YuriSizov) [inline editor/code_editor.cpp:130]: I don't think it's safe to remove this kind of check. But it seems that for some reason the check is limited to the line edit. I think we can simply check if the text editor is in focus or _any control_ within the FindReplaceBar is, not just the line edit.

**Review by YuriSizov — CHANGES_REQUESTED** [review]

**Review by KoBeWi — APPROVED** [review]

**Comment by akien-mga** [comment]
Could you rebase to get rid of the merge commit?

Also, the commit message "Fixed: ESC key not closing the FindReplaceBar" and the PR title "Refocus Code Editor after pressing ReplaceAll button" don't match, I'm not sure which is more accurate (the former of the PR title is better though). Could be worth clarifying.

File: editor/code_editor.cpp [hunk 4]
```diff
@@ -126,7 +126,7 @@ void FindReplaceBar::unhandled_input(const Ref<InputEvent> &p_event) {
 	if (k.is_valid() && k->is_action_pressed(SNAME("ui_cancel"), false, true)) {
 		Control *focus_owner = get_viewport()->gui_get_focus_owner();
 
-		if (text_editor->has_focus() || (focus_owner && vbc_lineedit->is_ancestor_of(focus_owner))) {
+		if (text_editor->has_focus() || (focus_owner && this->is_ancestor_of(focus_owner))) {
```
Comment (KoBeWi) [inline editor/code_editor.cpp:129]:
```suggestion
		if (text_editor->has_focus() || (focus_owner && is_ancestor_of(focus_owner))) {
```

**Comment by KoBeWi** [comment]
The PR title is outdated and doesn't describe the changes anymore.

**Comment by stmSi** [comment]
@KoBeWi @akien-mga @YuriSizov I made a new PR #79079 .... I don't what went wrong with my current local branch... It is very messy...

**Pull request closed without merging**